src/controllers/trust.information.controller.ts

```diff
@@ -22,8 +25,84 @@ export const get = (req: Request, res: Response, next: NextFunction) => {
   }
 };
 
-export const post = (req: Request, res: Response) => {
-  logger.debugRequest(req, `POST ${config.TRUST_INFO_PAGE}`);
+export const post = (req: Request, res: Response, next: NextFunction) => {
+  try {
+    logger.debugRequest(req, `POST ${config.TRUST_INFO_PAGE}`);
+
+    const beneficialOwnerIDs: string[] = [];
+    if (typeof req.body.beneficialOwners === 'string') {
+      beneficialOwnerIDs.push(req.body.beneficialOwners);
+
+    } else {
+      for (const bo in req.body.beneficialOwners) {
+        beneficialOwnerIDs.push(req.body.beneficialOwners[bo]);
+      }
+    }
+
+    const obj: trustType.Trust[] = JSON.parse(req.body.trusts);
+    const t: trustType.Trusts = {
+      trusts: obj
+    };
+
+    // Generate unique trust_id for each trust
+    let trustCount = 0;
+    if (req.session?.data.extra_data.roe.trusts !== undefined) {
+      trustCount = (req.session?.data.extra_data.roe.trusts)?.length;
+    }
+
+    const trustIDs: string[] = [];
+    for (const i in t.trusts) {
+      trustCount++;
+      t.trusts[i].trust_id = trustCount.toString();
+      trustIDs.push(trustCount.toString());
+    }
+
+    setTrustIDs(req, beneficialOwnerIDs, trustIDs);
+
+
+    const data: ApplicationDataType = setTrustData(t);
+
+    for (const i in data[TrustKey]) {
+      setApplicationData(req.session, data[TrustKey][i], TrustKey);
+    }
 
-  return res.redirect("TODO");
+    if (req.body.add) {
+      return res.redirect(config.TRUST_INFO_URL);
+    }
+    if (req.body.submit) {
+      return res.redirect(config.CHECK_YOUR_ANSWERS_PAGE);
+    }
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+const setTrustData = (obj: any): ApplicationDataType => {
+  const data: ApplicationDataType = prepareData(obj, TrustKeys);
+  return data;
+};
+
+// set Trust IDs to each selected Beneficial Owner
+const setTrustIDs = (req: any, beneficialOwnerIDs: string[], trustIDs: string[]) => {
+  for (const i in beneficialOwnerIDs) {
+    const individual = getFromApplicationDataIfPresent(req, BeneficialOwnerIndividualKey, beneficialOwnerIDs[i]);
+    if (individual !== undefined) {
+      for (const j in trustIDs) {
+        if (individual.trust_ids === undefined) {
```

Comment:
Is this check and pre-initialisation needed? Can't you just do the 'push' when no list yet exists?

Reply:
Yes, it didn't seem to work without this check.